runner/scheduler.py

```diff
@@ -120,19 +119,31 @@ def reset_time(self):
             self.traveller.stop()
             self.traveller = None
 
-    def set_to_wait(self, unique_id: str, action: WaitAction, log_this: bool = True):
+    def set_to_wait(self, example: TestExample, action: WaitAction, log_this: bool = True):
+        unique_id = example.unique_id
+
+        # We convert the percentage to tokens
+        if action.percentage_finished > 0.0:
+            span_percent = example.dataset_generator.memory_span / 100
+            percent_as_tokens = span_percent * action.percentage_finished
+
+            # Compute at what token we are at relative to this test's memory span, then calculate the difference.
+            tokens_elapsed = self.total_token_count - example.start_token
+            token_wait = percent_as_tokens - tokens_elapsed
+
+        else:
+            token_wait = action.tokens
```

Comment:
This might override `action.tokens` and could also result in a negative token count. It would be safer to take the `max(0, percent_as_tokens - tokens_elapsed, action.tokens)`.